pkg/booleanpolicy/querybuilders/special_cases.go

```diff
@@ -131,3 +131,15 @@ func mapFixedByValue(s string) string {
 	}
 	return valueToStringRegex(s)
 }
+
+// ForImageSignatureVerificationStatus returns a query builder for Image
+// Signature Verification Status.
+func ForImageSignatureVerificationStatus() QueryBuilder {
+	qbf := func(group *storage.PolicyGroup) []*query.FieldQuery {
+		group.Negate = !group.Negate
```

Comment:
I don't think this is a good idea here.
We are changing the value of the `*storage.PolicyGroup`. Instead, I'd rather vote to change the `negate` value of the `query.FieldQuery`.

If we keep it as such, we will observe the following when creating a policy:
When you first create a policy, in the `Preview violations` section you will observe the "not-negated" alert, meaning the alert for the image that has _is verified_.
This is due to the fact that `negate` will be set to `false` when first evaluating the policy during dry-run job.
![Screenshot 2022-04-05 at 04 38 02](https://user-images.githubusercontent.com/89904305/161668235-092e2e1c-eda9-4325-aade-b22076b45c17.png)
This will _also_ affect the violations of sensor. It will _still_ give the inverted violation for the image that is signed and verified, instead of the images that are not.


When you re-open the previously created policy afterwards and checking out the `Preview violations` section, the alerts are now created as expected, and `negated` is set to `true`.
![Screenshot 2022-04-05 at 04 41 04](https://user-images.githubusercontent.com/89904305/161668453-3d28369a-d6d1-4e03-a434-75b2c2a6fce1.png)

You would have to change the policy once more, only _afterwards_ the group's negate value will be taken into account and the violations are now correctly "negated" and show images that either don't have a signature or an unverifiable signature.

I'm not sure how to better change this within the unit tests, but either way we need to fix this before merging.

Reply:
You're right, thanks.